### PR TITLE
Use profile markers API for in-game profiling

### PR DESCRIPTION
**Pull request by zeroZshadow**
Move View.ApplyDiff to outside Worker.Tick

#### Description
This changes the use of the Profiler API in runtime code to use the ProfileMarkers.
This allows them to be used for measuring performance in the benchmark framework, and are lighter on the network when profiling remotely.

As a bonus, this moves View.ApplyDiff to its rightfull place amongst other ApplyDiff function calls.

### CODE REVIEW

**Comment by zeroZshadow** [comment]
This should be WIP, but I mis clicked 🗡

File: workers/unity/Packages/io.improbable.gdk.core/Systems/EcsViewSystem.cs [hunk 1]
```diff
@@ -16,6 +16,10 @@ public class EcsViewSystem : ComponentSystem
             new Dictionary<uint, IEcsViewManager>();
 
         private WorkerSystem worker;
+        
+        private ProfilerMarker applyDiffMarker = new ProfilerMarker("EcsViewSystem.ApplyDiff");
+        private ProfilerMarker addEntityMarker = new ProfilerMarker("EcsViewSystem.OnAddEntity");
+        private ProfilerMarker removeEntityMarker = new ProfilerMarker("EcsViewSystem.OnRemoveEntity");
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.core/Systems/EcsViewSystem.cs]: If would be neat if these could be annotations instead given we just wrap the entire method body in the using block. 

Unity pls

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs [hunk 2]
```diff
@@ -123,14 +123,9 @@ public void Return(string toReturn)
             Line($"return {toReturn};");
         }
 
-        public void ProfilerStart(string name)
+        public void ProfileScope(string markerVariableName, Action<CustomScopeBlock> populate)
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs]: :rip: @paulbalaji's docs

Comment (paulbalaji) [reply workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs]: it's fine - only a few LoC docs change required

File: workers/unity/Packages/io.improbable.gdk.core/Systems/EcsViewSystem.cs (hunk 1 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.gdk.core/Systems/EcsViewSystem.cs]: Its on my TODO list once we have the IL post processor

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs (hunk 2 above)
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs]:
```suggestion
        public void ProfileScope(string markerVariableName, Action<CustomScopeBlock> populate)
        {
            Add(new CustomScopeBlock($"using ({markerVariableName}.Auto())", populate));
        }

        public void ProfileScope(string markerVariableName, Func<IEnumerable<string>> populate)
```

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs [hunk 3]
```diff
@@ -123,14 +123,14 @@ public void Return(string toReturn)
...
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGenerationLib/CodeWriter/Scopes/Abstract/ScopeBody.cs]: Need to update the codegen docs where we have examples of using these 

https://github.com/spatialos/gdk-for-unity/blob/feature/profile-markers/workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md#profiler-sampling

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md [hunk 4]
```diff
@@ -570,31 +570,66 @@ using (var x = Some.RandomThing())
...
 
 ```csharp
+private ProfileMarker marker = new ProfileMarker("some name");
+
 public void ExampleMethod()
 {
-    Profiler.BeginSample("Sample name");
+    using (marker.Auto())
+    {
+        // define code here
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md]:
```suggestion
        // code here
```

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md [hunk 5]
```diff
@@ -570,31 +570,66 @@ using (var x = Some.RandomThing())
...
+
+```csharp
+
+t.Line("private ProfileMarker marker = new ProfileMarker(\"some name\");");
+
+t.Method("public void ExampleMethod()", m =>
+{
+    m.ProfilerScope("marker", () => {
+        yield return "//code goes here";
+        yield return "//more code";
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md]:
```suggestion
        yield return "// define code here";
        yield return "// define more code here";
```

File: workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md [hunk 6]
```diff
@@ -570,31 +570,66 @@ using (var x = Some.RandomThing())
...
+
+```csharp
+private ProfileMarker marker = new ProfileMarker("some name");
+
+public void ExampleMethod()
+{
+    using (marker.Auto())
+    {
+        // code goes here
+        // more code
```
Comment (paulbalaji) [inline workers/unity/Packages/io.improbable.gdk.tools/.CodeGenDocs/3-using-the-codewriter.md]:
```suggestion
        // code here
        // more code here
```

**Pull request merged**